hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneFsDelete.java

```diff
@@ -139,10 +140,25 @@ protected void processPath(PathData item) throws IOException {
       if (moveToTrash(item) || !canBeSafelyDeleted(item)) {
         return;
       }
-      if (!item.fs.delete(path, deleteDirs)) {
+      BasicRootedOzoneFileSystem.InnerDeleteResult innerDeleteResult = null;
+      if (item.fs.getUri().getScheme().equals(OZONE_OFS_URI_SCHEME)) {
+        BasicRootedOzoneFileSystem ofs = (BasicRootedOzoneFileSystem) item.fs;
```

Comment:
Is it ensured that if its OFS URI, internal implementation is BasicRootedOzoneFileSystem ?

Reply:
Yes

Reply:
I mean do we need add check item.fs instanceof BasicRootedOzoneFileSystem to ensure for type safety

Reply:
No , the fs instance will mostly be of class RootedOzoneFileSystem. Also hadoop2 and hadoop3 have different package for the same class, checking URI scheme should avoid problems like these.

Reply:
2 problem in this case,
1. BasicRootedOzoneFileSystem directly provide deleteKeys, not a part of FileSystem interface, so this can not be extended
2. other URI type is not handled, like o3fs or others